src/librustc/hir/mod.rs

```diff
@@ -725,6 +725,7 @@ pub struct ModuleItems {
 pub struct Crate {
     pub module: Mod,
     pub attrs: HirVec<Attribute>,
+    pub proc_macros: HirVec<DefId>,
```

Comment:
Not sure you need this, you can rely entirely on attributes, I think?

Reply:
Do you mean checking for the ``#[proc_macro]` attribute on the function? That would require us to repeatedly search through all of the functions. Also, I don't think the metadata code should care about the specific attributes that mark a function as being a proc macro.

Reply:
Okay so we can defer this.